### PR TITLE
Researcher account page

### PR DESCRIPTION
**Pull request by chrisbendel**

### CODE REVIEW

File: backend/.rubocop.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ Layout/EmptyLinesAroundBlockBody:
 
 Layout/LineLength:
   Max: 100
-  IgnoredPatterns: ['(\A|\s)#']
+  AllowedPatterns: ['(\A|\s)#']
```
Comment (chrisbendel) [inline backend/.rubocop.yml:20]: Rubocop suggested updating these

File: backend/db/structure.sql [hunk 2]
```diff
@@ -145,8 +145,8 @@ CREATE TABLE public.activesupport_cache_entries (
 CREATE TABLE public.admins (
     id bigint NOT NULL,
     user_id uuid NOT NULL,
-    created_at timestamp(6) without time zone NOT NULL,
-    updated_at timestamp(6) without time zone NOT NULL
+    created_at timestamp(6) with time zone NOT NULL,
```
Comment (chrisbendel) [inline backend/db/structure.sql:148]: I have no idea why this keeps happening with my migrations!?

File: backend/app/controllers/development/users_controller.rb [hunk 3]
```diff
@@ -7,12 +7,48 @@ class Development::UsersController < ApplicationController; end
...
+      last_name: 'Tres'
+    },
+    {
+      user_id: '00000000-0000-0000-0000-000000000005',
+      role: 'user',
+      name: 'User Cuatro',
+      first_name: 'User',
+      last_name: 'Cuatro'
+    }
   ].freeze
```
Comment (nathanstitt) [inline backend/app/controllers/development/users_controller.rb:52]: hrmm.. I find the earlier version much easer to read. is there a reason you expanded all of them?

File: backend/app/controllers/development/users_controller.rb [hunk 4]
```diff
@@ -40,38 +76,77 @@ def log_out
...
+    users[:researchers] ||= researchers
+    users[:admins] ||= admins
     users[:users] = MOCK_USERS.filter { |u| u[:role] == 'user' }
     render json: users, status: :ok
   end
 
   def ensure_users_exist
-    Admin.find_or_create_by(user_id: '00000000-0000-0000-0000-000000000000')
+    Admin.find_or_create_by(user_id: '00000000-0000-0000-0000-000000000001')
     Researcher.find_or_create_by(user_id: '00000000-0000-0000-0000-000000000001')
```
Comment (nathanstitt) [inline backend/app/controllers/development/users_controller.rb:87]: why change admin's uuid?  seems to me that admin should have lowest id at `0`?

File: frontend/src/screens/researcher/researcher-profile.tsx [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+            </Modal>
+        </Box>
+    )
+}
+
+const ProfileForm: React.FC<{className?: string}> = ({ className }) => {
+    const api = useApi()
+    const [researcher, setResearcher] = useState(useCurrentResearcher())
+    if (!researcher) {
+        return <></>
```
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-profile.tsx:234]: Yeah this change had to do with the setting of the avatar - the update_avatar endpoint wasn't returning the updated URL, so I modified the controller to return the researcher object with the new URL, and then when we get the response we update the state here so the picture "updates" on the form right after they've uploaded it

File: frontend/src/screens/researcher/researcher-profile.tsx [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+            width: 130,
+            justifyContent: 'center',
+        },
+        '.form-control': {
+            backgroundColor: editing ? 'transparent' : `${colors.lightGray} !important`,
+        },
+        '.select': {
+            backgroundColor: editing ? 'transparent' : `${colors.lightGray} !important`,
+        },
+    }
```
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-profile.tsx:259]: The one thing I needed here was access to that `editing` value, I tried looking into the `styled` docs but couldn't find how to define it outside of the component and then essentially pass a parameter in, is there an example of doing that somewhere else in the project?

File: frontend/src/screens/researcher/researcher-profile.tsx [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+
+                <Box className='col-3 pl-3'>
+                    <Resources direction='column' gap='small'>
+                        <ResourceLinks />
+                        <Box gap='medium' className='mt-1' align='center'>
+                            <CustomerSupportImage height={100} />
+                            <Box direction='column'>
+                                <HelpLink/>
+                            </Box>
+                        </Box>
```
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-profile.tsx:86]: <HelpLink/> has multiple children oddly enough, i forget what it's called but its like, so box is doing something here (<helplink> is displayed in rows in other places)
<>
  <thing1>
  <thing2>
</>

File: backend/app/controllers/development/users_controller.rb (hunk 3 above)
Comment (chrisbendel) [reply backend/app/controllers/development/users_controller.rb:52]: it was getting a bit long on each line (~125 chars with the leading whitespace) i will revert if needed, maybe we can add a max line length lint rule in the future?

File: frontend/src/screens/researcher/researcher-profile.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-profile.tsx:259]: ^ ignore the above 😆 i was totally overthinking this

File: frontend/src/screens/researcher/researcher-profile.tsx [hunk 8]
```diff
@@ -0,0 +1,422 @@
+import { Box, cx, Footer, HelpLink, Icon, InputField, ResourceLinks, SelectField, TopNavBar } from '@components';
+import { React, styled, useState } from '@common';
+import { useApi, useCurrentResearcher, useEnvironment } from '@lib';
+import { colors } from '../../theme';
+import { Researcher } from '@api';
+import { Button, Form, FormCancelButton, FormSaveButton, Select, Tooltip } from '@nathanstitt/sundry';
+import * as Yup from 'yup';
+import CustomerSupportImage from '../../components/customer-support-image';
+import RiceLogoURL from '../../images/rice-logo-darktext.png';
+import DefaultAvatar from '../../images/default-avatar.png';
```
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-profile.tsx:10]: I can check with Viki and come back to this - she's been using PNG's in favor of SVGs recently, im sure there is a reason, hopefully this isn't a hold up for merging though :D

**Comment by chrisbendel** [comment]
@nathanstitt hit all your notes - thank you!

Tests should all be good too, mostly just linting stuff 😃

File: frontend/src/screens/researcher/researcher-profile.tsx [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+                    </Box>
+                </Box>
+            </Col>
+            <Col sm={2}>
+                <a href='https://drive.google.com/file/d/1x1M8EcrOOu5U1ZQAtVmhvH3DkTlhtc8I/view' target='_blank'>
+                    <span>Check Details</span>
+                    <Icon icon="right" />
+                </a>
+            </Col>
+        </Box>
```
Comment (nathanstitt) [inline frontend/src/screens/researcher/researcher-profile.tsx:107]: This gets a little ugly on a narrowish (<900px) screen:
image

I think if you enclose them in `Row` , or just add `.row` to Box they'd wrap like the the Col should.

File: frontend/src/screens/researcher/researcher-profile.tsx [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+    maxHeight: 250,
+    border: `1px solid ${colors.lightGray}`,
+    marginTop: 40,
+    borderRadius: 5,
+    backgroundColor: colors.white,
+    padding: 20,
+    'a': {
+        color: colors.grayText,
+    },
+})
```
Comment (nathanstitt) [inline frontend/src/screens/researcher/researcher-profile.tsx:217]: we should remember these and export them if we need something similar in the future

**Review by nathanstitt — APPROVED** [review]
Looking good Chris, a few comments and need to fix the console warning

File: frontend/src/screens/researcher/researcher-account-form.tsx [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+        width: 130,
+        justifyContent: 'center',
+    },
+    '.form-control': {
+        backgroundColor: editing ? 'transparent' : `${colors.lightGray} !important`,
+    },
+    '.select': {
+        backgroundColor: editing ? 'transparent' : `${colors.lightGray} !important`,
+    },
+})
```
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-account-form.tsx:54]: ah yes - thank you, I tried this earlier and couldn't figure out the types 🤦

File: frontend/src/screens/researcher/researcher-account-form.tsx [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+                    onChange={validateCount}
+                />
+                <CharacterCount count={counts['lastName']} max={50} />
+            </div>
+
+            <div className='col-12 mt-1'>
+                <h6>Institution</h6>
+                <SelectField
+                    name="institution"
+                    isDisabled={!editing}
```
[the remark this answers is not in the document]
Comment (chrisbendel) [reply frontend/src/screens/researcher/researcher-account-form.tsx:144]: It looks like the parent form's `readOnly` takes care of this anyway, I was over-thinking it!

**Pull request merged**